lib/utils.nix

```diff
@@ -125,27 +125,185 @@ rec {
     end
   '';
 
+  /**
+    Generic "heredoc-style" escape utility.
+
+    # Args
+
+    - string: The string to escape.
+    - buildSurround: A function that maps Integer to Surround. Described in detail below.
+    - surroundWithNewlines: A boolean that impacts how to surround the string when escaping it.
+
+    A Surround is an attr set that looks like this:
+
+    ```nix
+    {
+      open = "(";
+      close = ")";
+    }
+    ```
+
+    `buildSurround` is a function that takes a depth Integer and returns a
+    Surround. For exmple, to build the previous Surround:
+
+    ```nix
+    nix-repl> buildSurround = depth: {
+      open = lib.strings.replicate depth "(";
+      close = lib.strings.replicate depth ")";
+    }
+    nix-repl> buildSurround 1
+    {
+      close = ")";
+      open = "(";
+    }
+    ```
+
+    We invoke `buildSurround` builder starting a depth 0, and increasing the
+    depth until we have a Surround where `surround.close` is not in the
+    string we're escaping.
+
+    We return the original string, first (possibly, depending on
+    `surroundWithNewlines`) surrounded by newlines , and secondly surrounded by
+    the Surround we found above.
+    ```
+  */
+  escapeString =
```

Comment:
I'm just realizing this is a pretty generic name for a very specific thing. A better name might be something like `heredocStyleEscapeString`. An even better thing to do might be to move this out of the `lib` attrset and into a let binding on top to emphasize that this is an implementation detail, and not meant to be used outside of nixvim.

Happy to make whatever changes here folks suggest.